src/util/input-types.ts

```diff
@@ -0,0 +1,10 @@
+import {Observable, ObservableInput} from '../Observable';
+
+export type _mapProject<T, R> = (value: T, index: number) => R;
```

Comment:
I tried this by myself and found it's indifferent to have `_` or not, by this are actually not being exposed as pubic type definition. If intention is to use idiom of non-exposed export, I'm ok with this. Up to you. :)